crates/node/src/cli/mod.rs

```diff
@@ -98,22 +126,68 @@ impl RunCmd {
     }
 
     pub fn get_config(&self) -> anyhow::Result<Arc<ChainConfig>> {
-        let mut chain_config: ChainConfig = match &self.preset {
-            Some(preset_name) => ChainConfig::from_preset(preset_name.as_str())?,
+        let chain_config = match &self.preset {
+            Some(preset_name) => ChainConfig::from_preset(preset_name.as_str()).map_err(|err| {
+                log::error!("Failed to load config from preset '{}': {}", preset_name, err);
+                anyhow::anyhow!(err)
+            })?,
             None => {
-                ChainConfig::from_yaml(&self.chain_config_path.clone().expect("Failed to retrieve chain config path"))?
+                let path = self.chain_config_path.clone().ok_or_else(|| {
+                    log::error!("{}", "Chain config path is not set");
+                    anyhow::anyhow!(if self.is_sequencer() {
+                        "In Sequencer or Devnet mode, you must define a Chain config path with `--chain-config-path <CHAIN CONFIG FILE PATH>` or use a preset with `--preset <PRESET NAME>`.\nThe default presets are:\n- 'mainnet' - (crates/primitives/chain_config/presets/mainnet.yml)\n- 'sepolia' - (crates/primitives/chain_config/presets/sepolia.yml)\n- 'integration' - (crates/primitives/chain_config/presets/integration.yml)\n- 'test' - (crates/primitives/chain_config/presets/test.yml)"
+                    } else {
+                        "No network specified. Please provide a network with `--network <NETWORK>` or a custom Chain config path with `--chain-config-path <CHAIN CONFIG FILE PATH>` or use a preset with `--preset <PRESET NAME>`.\nThe default presets are:\n- 'mainnet' - (crates/primitives/chain_config/presets/mainnet.yml)\n- 'sepolia' - (crates/primitives/chain_config/presets/sepolia.yml)\n- 'integration' - (crates/primitives/chain_config/presets/integration.yml)\n- 'test' - (crates/primitives/chain_config/presets/test.yml)"
+                    })
+                })?;
+                ChainConfig::from_yaml(&path).map_err(|err| {
+                    log::error!("Failed to load config from YAML at path '{}': {}", path.display(), err);
+                    anyhow::anyhow!(err)
+                })?
             }
         };
 
-        // Override stuff if flag is setted
-        if self.chain_config_override {
-            chain_config = self.chain_params.override_cfg(chain_config);
-        }
+        // Override stuff if flag is set
+        let chain_config =
+            if self.chain_config_override { self.chain_params.override_cfg(chain_config) } else { chain_config };
```

Comment:
As said in DMs, I've tried to implement a different override mechanism but it's far from perfect at the moment because `ChainConfig` doesn't implement `Serialize` (since our flow is _1._ generate `ChainConfig` _2._ override if possible , we need to convert the pre-built `ChainConfig` to `Value`).

Instead of going down the rabbit hole of implementing wrapper structs around Blockifiers one & implementing Serialize myself, I've created a wrapper struct `OverridableChainConfig` that represents the fields we're able to override.

I'm not really fan of this tbh but it works at least. Any thoughts? Implementing Serialize would be the cleanest probably but didn't try yet - can do if you judge it better